.github/scripts/version_updater.rb

```diff
@@ -0,0 +1,124 @@
+require 'rubygems'
+require 'bundler'
+require 'bundler/cli'
+
+
+
+class GemUpdater
+    def initialize
+        @versions = {}
+
+        # keep track of each integration and a list of their gemfiles that they are in
+        # "grpc"=>[["../gemfiles/ruby_2.5_contrib.gemfile", #<Gem::Version "1.48.0">], ["../gemfiles/ruby_2.6_contrib.gemfile", #<Gem::Version "1.48.0">]
+        @gems_in = Hash.new {|hash, key| hash[key] = [] }
+    end
+
+    # Run bundler and handle errors
+    def run_bundler_command(gemfile, command_args)
+        Bundler.reset!
+        ENV['BUNDLE_GEMFILE'] = gemfile
+        Bundler::CLI.start(command_args)
+    rescue StandardError => e
+        puts "Error: #{e.message}"
+    end
+
+    # Update a gem in all its gemfiles
+    def update_gem(gem_to_update)
+        # HACK - setting this to grpc while testing out the script
+        # gem_to_update = 'grpc'
+        @gems_in[gem_to_update].each do |gemfile, version|
+            puts "Checking to see if #{gem_to_update} needs to be updated in #{gemfile}"
+            gemfile_lock = "#{gemfile}.lock"
+            definition = Bundler::Definition.build(gemfile, gemfile_lock, nil)
+            dependencies = definition.dependencies
+
+            dependencies.each do |dep|
+                next if gem_to_update != dep.name # only update the gem we chose at random
+                # puts "Skipping #{dep.name} as it is not #{gem_to_update}"
+                
+                gem_name = dep.name
+                gem_version = dep.requirements_list
+                gem_requirements = Gem::Requirement.create(*gem_version)
+                gem_platforms = dep.platforms
+        
+                puts "Gem Name: #{gem_name}"
+                puts "Gem Version: #{gem_version}"
+                puts "Gem Requirments: #{gem_requirements}"
+                puts "Gem Platforms: #{gem_platforms}"
+        
+                latest = Gem.latest_version_for(gem_name)
```

Comment:
This `latest = Gem.latest_version_for(gem_name)` can actually be moved outside of this loop (should save some time)

```suggestion
        
            latest = Gem.latest_version_for(gem_name)
            dependencies.each do |dep|
                next if gem_to_update != dep.name # only update the gem we chose at random
                # puts "Skipping #{dep.name} as it is not #{gem_to_update}"
                
                gem_name = dep.name
                gem_version = dep.requirements_list
                gem_requirements = Gem::Requirement.create(*gem_version)
                gem_platforms = dep.platforms
        
                puts "Gem Name: #{gem_name}"
                puts "Gem Version: #{gem_version}"
                puts "Gem Requirments: #{gem_requirements}"
                puts "Gem Platforms: #{gem_platforms}"
```